pkg/controller/replicator.go

```diff
@@ -143,16 +145,42 @@ func (r *replicator) WriteAt(p []byte, off int64) (int, error) {
 	return n, err
 }
 
-func (r *replicator) buildReadWriters() {
+func (r *replicator) UnmapAt(length uint32, off int64) (int, error) {
+	if !r.backendsAvailable {
+		return 0, ErrNoBackend
+	}
+
+	n, err := r.unmapper.UnmapAt(length, off)
+	if err != nil {
+		errs := map[string]error{
+			r.unmapperIndex[0]: err,
```

Comment:
Should we need index check for `unmapperIndex[0]` or `unmapperIndex[index]` below to avoid any unexpected out of index issue? (I know you do the same as readerIndex and writeIndex)

Reply:
Ideally, the controller should error out once there is no available replica (while it is still running and receiving the requests)
But just in case, we can add it.